PG.Commons/PG.Commons/Util/StringUtility.cs

```diff
@@ -76,12 +82,21 @@ public static List<string> SplitClean(string s, char separator)
             return SplitStringInternal(s, separator);
         }
 
+        /// <summary>
+        /// Parses a given string into a list. Optionally cleans all strings by either
+        /// dropping whitespace-only/null strings, trimming them, or both. 
+        /// </summary>
+        /// <param name="s"></param>
+        /// <param name="separator"></param>
+        /// <param name="shouldClean"></param>
+        /// <returns></returns>
         [NotNull]
         public static List<string> ParseSeparatedStringToList(string s, char separator, bool shouldClean = true)
         {
+            // TODO: Add option to retain empty strings in the list.
             return SplitStringInternal(s, separator, shouldClean);
```

Comment:
`SplitStringInternal` should be implemeted using `Span<T>`. Looks very slow for such a common utility class. 
Btw, where did we "stole" that class from? Because you say  we are missing copyright notice here.

Reply:
Currently, there is no need to switch the implementation in my opinion:

1. We do not have performance issues or memory problems.
2. As far as my research goes, `string.Split(...)` is fairly optimized using unsafe pointer magic under the hood.

And I'm not sure I can come up with a better optimized `Split(...)` function and prove that it actually improves performance and/or reduces the memory footprint.

So if it is a problem thet you're adamant about I'd suggest opening an issue for it.

Reply:
probably only if we really need to improve this.